pallets/gear/src/internal.rs

```diff
@@ -780,6 +818,26 @@ where
         });
     }
 
+    pub(crate) fn send_message_from_reservation(
```

Comment:
that's not intuitive that inside it consumes the reservation (rename or remove this fn at all)